test/test_gax.py

```diff
@@ -256,20 +293,28 @@ def test_add_done_callback(self):
         operation_future = self._make_operation_future(
```

Comment:
I didn't catch this on the first PR, but this test doesn't establish that the `operation_future` actually waited for the operation to be done; that is, an adversarial `_OperationFuture` could pass this test by always executing callbacks immediately.

Could you add one more test that checks that the callbacks are not executed before the operation is done? A simple way to do this might be to set some fields of `Simple` when calling `_make_operation()`, and checking that the fields are set as expected in your callbacks.